spacy_llm/tests/test_cache.py

```diff
@@ -74,7 +74,9 @@ def test_caching(use_pipe: bool) -> None:
         nlp_2 = _init_nlp(tmpdir)
         [nlp_2(text) for text in texts]
         cache = nlp_2.get_pipe("llm")._cache  # type: ignore
-        assert cache._stats["hit"] == n
+        # The number of hits should be batch_size n * 2 since
+        # __contains__ and __getitem__ both log hits
+        assert cache._stats["hit"] == n * 2
```

Comment:
Not sure why we weren't getting a failure here before. But this is technically correct based on the impl of the cache. I don't really think this is how the stats should work though. Shouldn't these stats only increment in `__getitem__`? It's not super important

Reply:
That's because you're using one more `__contains__()` per doc in your rewrite than the original implementation (for `__call__()`), which changes the stats accordingly. As for how the stats should behave - how about we are more granular about this? We could count the number of hits in `__getitem__()` and those in `__contains__()` separately and call them `hits_getitem` and `hits_contains` (and maybe one `hits` counting all hits, regardless of how they happen).